Major fix makefile, Fixes issue #33

Tabrez-dev opened the pull request:
Fixes all issues in #33 
This includes fixes in minor-fix-makefile(previous PR on this issue).

1. Added Makefiles inside src directory and in its sub directories.
2. Added a Makefile in the rv32_asm directory to handle assembly build task.

Tabrez-dev committed:
Fix minor issues in Makefile for Release 0.1 (#33)

Tabrez-dev committed:
build on each folder

Tabrez-dev committed:
first working version

Tabrez-dev committed:
cleaned up src make files

Tabrez-dev committed:
added makefile in rv32_asm directory

RahulGoyal-tech commented on Makefile:
```diff
@@ -1,84 +1,87 @@
+endef
+
+# Minimum required GCC version
+GCC_MIN_VERSION := 11.0.0
```
should we add a seperate config.mk to avoid hardcoding this in this makefile and providing a seperate file to configure this and other hardcoded variables like Target Executable name, build dir name, out directory name, ASM directory name if some user want to define them with some other name?

Tabrez-dev replied:
is config.mk required? @akashkollipara

akashkollipara replied:
No. Have at top/start of makefile.

pranjalchanda08 commented:
@akashkollipara Please have a look on this

akashkollipara commented:
@Tabrez-dev you are repeating the same pieces of recipe in build.mk files. try to have it top level makefile and only gather paths from build.mk

akashkollipara reviewed (changes requested):
Simplify build.mk

Tabrez-dev committed:
Simplify build.mk files as per reviewer suggestions

- Removed repetitive compilation logic from build.mk files
- Moved compilation rules to top-level Makefile
- Kept build.mk files focused on source file path gathering

Tabrez-dev commented:
> Simplify build.mk

@akashkollipara I've simplified the build.mk files to only gather source file paths. 
The compilation rules are now centralized in the top-level Makefile. Can you please check it?

pranjalchanda08 commented:
@akashkollipara please have a look
